aten/src/ATen/native/TensorFactories.cpp

```diff
@@ -982,6 +982,34 @@ Tensor hann_window(
       window_length, periodic, /*alpha=*/0.5, /*beta=*/0.5, options);
 }
 
+// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~ kaiser_window ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Tensor kaiser_window(int64_t window_length, double beta, const TensorOptions& options) {
+  return native::kaiser_window(window_length, beta, /*periodic=*/true, options);
+}
+
+Tensor kaiser_window(
+    int64_t window_length,
+    double beta,
+    bool periodic,
+    const TensorOptions& options) {
+  window_function_checks("kaiser_window", options, window_length);
+  if (window_length == 0) {
+    return at::empty({0}, options);
+  }
+  if (window_length == 1) {
+    return native::ones({1}, options);
+  }
+  if (periodic) {
+    window_length += 1;
+  }
+  auto initial = native::arange(window_length, options);
+  auto window = native::zeros(window_length, options);
+  auto iter = TensorIterator::unary_op(window, initial);
+  kaiser_window_stub(iter.device_type(), iter, window_length, beta);
```

Comment:
Writing a custom kernel to fuse the operations seems like a performance win. Are there any other reasons you decided to write a custom kernel (vs implementing the window as a composite of existing Torch operations)?

Reply:
No, the main/only reason was for performance, as mentioned in the i0 PR.